File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def dfs(start):
    visited[start] = True
    print(start, end=" ")

    for i in graph[start]:
        if not visited[i]:
            dfs(i)

def bfs(start):
    q = deque([start])
    visited[start] = True
    while q:

        v = q.popleft()
        print(v, end=" ")
        for i in graph[v]:
            if not visited[i]:
                visited[i] = True
                q.append(i)

n, m, v = map(int, input().split())
graph = [[] for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

# 정렬
for i in graph:
    i.sort()

# dfs
visited = [False] * (n+1)
dfs(v)
print()

# bfs
visited = [False] * (n+1)
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;

public class Main {
    static StringBuilder sb = new StringBuilder();
    static int n, m, v;
    static boolean[] visitedDfs;
    static boolean[] visitedBfs;
    static Queue<Integer> q = new LinkedList<>();

    static ArrayList<ArrayList<Integer>> arr;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());
        visitedDfs = new boolean[n+1];
        visitedBfs = new boolean[n+1];
        arr = new ArrayList<>();
        // arr 초기화
        for (int i = 0; i<=n; i++) {
            arr.add(new ArrayList<>());
        }

        for (int i = 0; i<m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            arr.get(a).add(b);
            arr.get(b).add(a);
        }

        // 정렬과정
        for (int i = 1; i<=n; i++) {
            if (!arr.get(i).isEmpty()) {
                Collections.sort(arr.get(i));
            }
        }

        sb.append(v + " ");
        dfs(v);
        sb.append("\n");
        bfs(v);
        System.out.println(sb);

    }
    public static void dfs(int start) {
        visitedDfs[start] = true;
        for (int node: arr.get(start)) {
            if (!visitedDfs[node]) {
                sb.append(node + " ");
                visitedDfs[node] = true;
                dfs(node);
            }
        }
    }

    public static void bfs(int start) {
        q.add(start);
        visitedBfs[start] = true;
        sb.append(start + " ");

        while (!q.isEmpty()) {
            int node = q.poll();
            for (int no: arr.get(node)) {
                if (!visitedBfs[no]) {
                    visitedBfs[no] = true;
                    sb.append(no + " ");
                    q.add(no);
                }
            }

        }
    }
}